consensus/api/src/conversions.rs

```diff
@@ -71,63 +71,6 @@ impl From<Error> for ProposeTxResult {
     }
 }
 
-/// Convert ProposeTxResult --> TransactionValidationError.
-impl TryInto<Error> for ProposeTxResult {
```

Comment:
Removing this conversion is the main goal of this PR, since it helps develop #2484